### DISCUSSION TITLE
RememberMeListener

### DISCUSSION THREAD

**Discussion by meiyasan** [symfony/symfony#49003] [Q&A] [answered]
Hello ! I have been investigating why my RememberMe cookies are not generated.

I see from the profiler that the "RememberMeListener" in charge of creating this cookie is not in the list of events.. At this point I cannot figure out why.. 
Is there any obvious reason why this listener would not be in this list ? (Disabled?)

**Comment by MatTheCat** [comment]
Hello!

The listener is registered in [`RememberMeFactory::createAuthenticator`](https://github.com/symfony/symfony/blob/1f7bc1098f0b03181095d35b11fa48523864e6f2/src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/RememberMeFactory.php#L100-L104), but it won’t be called if none of your firewalls configure `remember_me`.

Else we’ll need a [reproducer](https://symfony.com/doc/current/contributing/code/reproducer.html) in order to investigate.

**Comment by meiyasan** [comment]
Hello @MatTheCat 

Here is a minimal repository as a reproducer
https://github.com/xKZL/symfony_issue_49003

The User entity is set and the firewall is the default one, with "remember_me" functionality enabled. 
I cannot see any listener set (either called, or not called) that might create a RememberMe token.
<img width="1022" alt="Screenshot 2023-01-17 at 22 41 28" src="https://user-images.githubusercontent.com/37673095/212914653-07a21fc8-2019-40cd-9f74-6fd3a7ba50ad.png">

**Reply by MatTheCat** [reply]
Thanks! (Please reply using “Write a reply” instead of “Suggest an answer” next time :wink:)

So it appears the `RememberMeListener` really is called (checked using xdebug) but it doesn’t appear in the profiler, which I think is related to https://github.com/symfony/symfony/issues/42973

**Comment by meiyasan** [comment]
Oops, sorry about that suggested answer :)
Ok so this is definitely a bug at least in the profiler display.

I actually also tried to traceback the issue, but in my case there is no way the RememberMeListener onSuccessLogin is called.
In my opinion it is like the RememberMeFactory is called, then a new listener is defined (one per firewall), but this is never called during login time.

I have added a /register route to create a dummy user. You can try to /login then, the "always remember me" option is enabled, but no token, and no call to RememberMeListener for the corresponding firewall

**Comment by meiyasan** [comment, marked as answer]
I eventually manager to properly set the rememberme cookie on the basic test case.
Sorry for bothering, but indeed https://github.com/symfony/symfony/issues/42973 issue is definitely very misleading... idk how to fix.
